pkg/tsdb/influxdb/response_parser_test.go

```diff
@@ -3,233 +3,188 @@ package influxdb
+		result = parser.Parse(response, query)
 
-					So(result.Series[0].Name, ShouldEqual, "alias 10m 10m")
-				})
+		require.Equal(t, result.Series[0].Name, "alias 10m 10m")
 
-				Convey("column alias", func() {
-					query := &Query{Alias: "alias $col", Measurement: "10m"}
-					result := parser.Parse(response, query)
+		query = &Query{Alias: "alias $col", Measurement: "10m"}
+		result = parser.Parse(response, query)
 
-					So(result.Series[0].Name, ShouldEqual, "alias mean")
-					So(result.Series[1].Name, ShouldEqual, "alias sum")
-				})
+		require.Equal(t, result.Series[0].Name, "alias mean")
+		require.Equal(t, result.Series[1].Name, "alias sum")
 
-				Convey("tag alias", func() {
-					query := &Query{Alias: "alias $tag_datacenter"}
-					result := parser.Parse(response, query)
+		query = &Query{Alias: "alias $tag_datacenter"}
+		result = parser.Parse(response, query)
 
-					So(result.Series[0].Name, ShouldEqual, "alias America")
-				})
+		require.Equal(t, result.Series[0].Name, "alias America")
 
-				Convey("segment alias", func() {
-					query := &Query{Alias: "alias $1"}
-					result := parser.Parse(response, query)
+		query = &Query{Alias: "alias $1"}
+		result = parser.Parse(response, query)
 
-					So(result.Series[0].Name, ShouldEqual, "alias upc")
-				})
+		require.Equal(t, result.Series[0].Name, "alias upc")
 
-				Convey("segment position out of bound", func() {
-					query := &Query{Alias: "alias $5"}
-					result := parser.Parse(response, query)
+		query = &Query{Alias: "alias $5"}
+		result = parser.Parse(response, query)
 
-					So(result.Series[0].Name, ShouldEqual, "alias $5")
-				})
-			})
+		require.Equal(t, result.Series[0].Name, "alias $5")
 
-			Convey("[[]] alias", func() {
-				Convey("simple alias", func() {
-					query := &Query{Alias: "series alias"}
-					result := parser.Parse(response, query)
+		query = &Query{Alias: "series alias"}
+		result = parser.Parse(response, query)
 
-					So(result.Series[0].Name, ShouldEqual, "series alias")
-				})
+		require.Equal(t, result.Series[0].Name, "series alias")
 
-				Convey("measurement alias", func() {
-					query := &Query{Alias: "alias [[m]] [[measurement]]", Measurement: "10m"}
-					result := parser.Parse(response, query)
+		query = &Query{Alias: "alias [[m]] [[measurement]]", Measurement: "10m"}
+		result = parser.Parse(response, query)
 
-					So(result.Series[0].Name, ShouldEqual, "alias 10m 10m")
-				})
+		require.Equal(t, result.Series[0].Name, "alias 10m 10m")
 
-				Convey("column alias", func() {
-					query := &Query{Alias: "alias [[col]]", Measurement: "10m"}
-					result := parser.Parse(response, query)
+		query = &Query{Alias: "alias [[col]]", Measurement: "10m"}
+		result = parser.Parse(response, query)
 
-					So(result.Series[0].Name, ShouldEqual, "alias mean")
-					So(result.Series[1].Name, ShouldEqual, "alias sum")
-				})
+		require.Equal(t, result.Series[0].Name, "alias mean")
+		require.Equal(t, result.Series[1].Name, "alias sum")
```

Comment:
Unless this test case all of these tests could have been table-driven like for example 
https://github.com/grafana/grafana/blob/f7d079c0a54aebf020d5da767900a6cdb64b6332/pkg/tsdb/interval/interval_test.go#L16-L32

Maybe it could if you have a slice of expected series names.

Reply:
Tried to, but it only really adds many lines of code and doesn't help much in this case I think, since we have to iterate through Series -> [][]Points etc. Good to keep in mind though for more suitable cases 👍